sphinx_gallery/scrapers.py

```diff
@@ -204,15 +204,25 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
             )
         )
 
+    # Determine whether single-img should be converted to multi-img
+    convert_to_multi_image = True  # default is to convert
+    if block_vars.get("multi_image") is not None:  # block setting takes precedence
+        convert_to_multi_image = block_vars["multi_image"] != "single"
+    elif block_vars["file_conf"].get("multi_image") is not None:  # then file setting
+        convert_to_multi_image = block_vars["file_conf"]["multi_image"] != "single"
+
     plt.close("all")
     rst = ""
     if len(image_rsts) == 1:
         rst = image_rsts[0]
     elif len(image_rsts) > 1:
-        image_rsts = [
-            re.sub(r":class: sphx-glr-single-img", ":class: sphx-glr-multi-img", image)
-            for image in image_rsts
-        ]
+        if convert_to_multi_image:
```

Comment:
I may have mis-read, but should there be an `else` for when `convert_to_multi_image` is True?

Reply:
So the existing behaviour for when there are multiple images from a single cell is to switch their CSS class from `single-img` (100% max width) to `multi-img` (42% max width). Then afterwards the images are added to a list.

With these changes, `convert_to_multi_image` is default `True` (so the class gets switched from `single-img` to `multi-img`), but if `sphinx_gallery_multi_image` or `sphinx_gallery_multi_image_block` is `"single"`, then `convert_to_multi_image` becomes `False` and the images remain as a `single-img` class. Regardless of whether the image class changes, they still get added to a list afterwards.

Based on the examples in tinybuild this gives the expected behaviour, e.g.:
- File-wide:
![image](https://github.com/user-attachments/assets/5a403d88-7653-4ffb-9342-a8af8aa4f099)

- Block-specific:
![image](https://github.com/user-attachments/assets/451fd16b-8109-4847-8e28-aed15ed83e79)

Reply:
Thanks mis-read the code